kong/plugins/statsd/log.lua

```diff
@@ -315,19 +402,31 @@ local function log(conf, messages)
       local metric_config_name = metric_config.name
       local metric = metrics[metric_config_name]
 
-      local name = get_scope_name(conf, message, metric_config.service_identifier or conf.service_identifier_default)
+      local name
+      local tags
+      if conf.tag_style then
+        tags = get_tags(conf, message, metric_config)
+      else
+        name = get_scope_name(conf, message, metric_config.service_identifier or conf.service_identifier_default)
+      end
 
       if metric then
-        metric(name, message, metric_config, logger, conf)
+        metric(name, message, metric_config, logger, conf, tags)
 
       else
         local stat_name = stat_name[metric_config_name]
         local stat_value = stat_value[metric_config_name]
 
         if stat_value ~= nil and stat_value ~= -1 then
-          logger:send_statsd(name .. "." .. stat_name, stat_value,
-                             logger.stat_types[metric_config.stat_type],
-                             metric_config.sample_rate)
+          if conf.tag_style then
+            logger:send_statsd(stat_name, stat_value,
+              logger.stat_types[metric_config.stat_type],
+              metric_config.sample_rate, tags, conf.tag_style)
```

Comment:
```suggestion
              metric_config.sample_rate, tags, conf.tag_style)

```

Please fix familiar places.